pkgs/development/haskell-modules/non-hackage-packages.nix

```diff
@@ -7,6 +7,8 @@
 # files.
 self: super: {
 
+  ldgallery-compiler = self.callPackage ../../tools/graphics/ldgallery/compiler { };
+
```

Comment:
The program is a command line tool and not a library, so I'm not sure whether it belongs to Hackage, or even to this listing. Is it sensible to put it in "non-hackage-packages"?

Reply:
It is easiest for us if it is on Hackage.

It if was on Hackage, then you wouldn't need the `pkgs/tools/graphics/ldgallery/compiler/default.nix`, since the Haskell automation in nixpkgs would keep it up to date automatically.

Hackage provides hosting for quite a few command line tools that don't provide a library.

Reply:
It turns out that the "compiler" part of ldgallery is not really usable without the "viewer" part which requires NodeJS to build.
Both versions also need to be kept in sync for the package to work.
This prevents ldgallery-compiler from being publishable to Hackage alone.